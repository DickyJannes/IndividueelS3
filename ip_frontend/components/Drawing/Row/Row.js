/* eslint-disable react/prop-types */
import React from "react";
import { View } from "react-native";

import Pixel from "../Pixel/Pixel";
import rowStyles from  "./rowStyle";

export default function Row(props) {
	const {width, selectedColor, newBackground} = props;

	let pixels = [];

	for (let i = 0; i < width; i++) {
		pixels.push(<Pixel key={i} selectedColor={selectedColor} newBackground={newBackground}/>);
	}

	return <View style={rowStyles.row}>{pixels}</View>;
}