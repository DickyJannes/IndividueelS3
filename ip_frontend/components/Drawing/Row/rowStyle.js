import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
	row: {
		display: "flex",
		width: "fit-content",
		flex: 1,
		flexDirection: "row",
	}
});

export default styles;
