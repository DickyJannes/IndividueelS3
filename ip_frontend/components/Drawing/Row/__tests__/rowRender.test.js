import React from 'react';
import renderer from 'react-test-renderer';

import Row from '../Row';

test('renders row correctly', () => {
  const tree = renderer.create(<Row />).toJSON();
  expect(tree).toMatchSnapshot();
});