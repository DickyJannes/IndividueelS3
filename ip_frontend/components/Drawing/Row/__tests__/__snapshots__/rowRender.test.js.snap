// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders row correctly 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "width": "fit-content",
    }
  }
/>
`;
